editor/editor_help.cpp

```diff
@@ -2935,7 +2948,65 @@ void EditorHelp::_gen_extensions_docs() {
 	}
 }
 
-void EditorHelp::generate_doc(bool p_use_cache) {
+void EditorHelp::_load_script_doc_cache() {
+	ERR_FAIL_COND_MSG(!ProjectSettings::get_singleton()->is_project_loaded(), "Error: cannot load script doc cache without a project.");
+	if (ResourceLoader::exists(get_script_doc_cache_full_path())) {
+		Ref<Resource> script_doc_cache_res = ResourceLoader::load(get_script_doc_cache_full_path());
+		if (script_doc_cache_res.is_valid()) {
+			Array classes = script_doc_cache_res->get_meta("classes", Array());
+			for (int i = 0; i < classes.size(); i++)
+				doc->add_doc(DocData::ClassDoc::from_dict(classes[i]));
+		}
+		return;
+	}
+
+	// Force compile all scripts, which should generate their documentation.
+	WARN_PRINT("Script documentation cache not found. Regenerating it may take a while.");
+	HashMap<String, bool> is_extension_script; // Caches ScriptServer results so it's not being queried for every file, only for every extension.
+	List<Ref<DirAccess>> dirs;
+	dirs.push_front(DirAccess::create(DirAccess::ACCESS_RESOURCES));
+	while (!dirs.is_empty()) {
+		Ref<DirAccess> dir = dirs.front()->get();
+		dirs.pop_front();
+		for (const String &file : dir->get_files()) {
+			if (file.begins_with("."))
+				continue;
+			if (!is_extension_script.has(file.get_extension()))
+				is_extension_script[file.get_extension()] = ScriptServer::get_language_for_extension(file.get_extension()) != nullptr;
+
+			if (is_extension_script[file.get_extension()]) {
+				Ref<Script> scr = ResourceLoader::load(dir->get_current_dir() + "/" + file);
```

Comment:
`dir->get_current_dir().path_join(file)` is preferable then using ` + "/" + `, that way it will concatene with "\" on Windows.